### ISSUE TITLE
How to exclude things from being weaved

### ISSUE THREAD

**Issue by kampilan** [ArxOne/MrAdvice#162]
First off, I have to congratulate you on a fantastic product. I just works great. I am using it to add an automatic logging feature to my logging framework. 

I can't seam to figure out how to exclude things from automatic logging. Specifically I am trying to prevent the logging of getters and setters of automatic properties and default constructors. These things create a lot of clutter with no instrumentation value. I have searched through the test but I just can't seem to find an example of how to use ExcludePointcut. What patterns are supported? Where to place the attribute. On the Advice or the Class or Assembly etc?

Thanks
Jim

**Comment by kampilan** [comment]
OK I dug through the closed issues and mostly answered my own question.

Facing one last issue .. exclude default constructors. Any ideas?

Thanks

**Comment by kampilan** [comment]
If I want to distribute my Advice in a nuget package but I don't want any weaving to occur when the package is built.

1. How can I prevent MrAdvice from weaving when by nuget package is built? I only want to reference interfaces and attributes.

2. Will the build tasks be added when someone adds my package, which references MrAdvice to their, project?

Thanks so much
Jim

**Comment by picrap** [comment]
Hi, sorry for ny rate answer. Currently there is no way to exclude constructors, we could open an issue for these.

**Comment by kampilan** [comment]
Thanks for the reply.
From my own selfish use case perspective:

1. Excluding automatic property setters and getters
2. Excluding default constructors
3. Excluding based on an attribute I define. My advice is called AutoWatch and having a NoAutoWatch that excludes when placed on classes and methods is clearer. I currently use a NoAutoWatch attribute to perform the exclude in my Advice (post weaving). It would be better to have the NoAutoWatch exclude from weaving altogether.

Thanks for considering these.
Jim

**Comment by kampilan** [comment]
Any chance of these items being considered?

Thanks
Jim

**Comment by picrap** [comment]
Yes, unfortunately I don’t work much for MrAdvice at the moment. This is probably quite simple to implement, so a pull request could be appreciated 😉

**Comment by kampilan** [comment]
Thanks for the status update. You should probably mention that you are no longer actively working on this project on the front page. A project like this is quite complex and the value I derive from it compared to the effort it would take me to figure out how to make those enhancements on my own says I should find something else. Good luck!

**Comment by picrap** [comment]
I think we’ll probably go with advising from code at build-time, so anyone will be able to perform its own customizations.
